spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryTracker.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.autoconfigure.telemetry;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.azure.management.Azure;
+import org.springframework.lang.NonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.microsoft.azure.spring.cloud.autoconfigure.telemetry.TelemetryProperties.*;
+
+public class TelemetryTracker {
+
+    private static final String PROJECT_VERSION = TelemetryTracker.class.getPackage().getImplementationVersion();
+
+    private static final String PROJECT_INFO = "spring-cloud-azure" + "/" + PROJECT_VERSION;
+
+    private TelemetryClient client;
+
+    private Azure azure;
+
+    public TelemetryTracker(Azure azure) {
+        this.client = new TelemetryClient();
+        this.azure = azure;
+    }
+
+    public void trackEvent(@NonNull String name, Map<String, String> customProperties) {
```

Comment:
If this is only used inside this class, could we make it private and customProperties as `@NonNull`?

Reply:
I am not sure if we need to provide more than one track function in this class. Or just keep one public method is enough here.